tests/test_load.py

```diff
@@ -23,3 +30,72 @@ def test_can_load_custom_yaml_data(monkeypatch, tmpdir):
     data = load_data()
 
     assert data == {"hello": "yaml"}
+
+
+@pytest.mark.parametrize(
+    "file_name, file_contents, expected",
+    [
+        ("myfile.json", '{"french": "toast"}', {"french": "toast"}),
+        ("myfile.yaml", "french: fries", {"french": "fries"}),
+    ],
+)
+def test_can_load_local_data_from_source(tmpdir, file_name, file_contents, expected):
+    json_file = tmpdir.join(file_name)
+    json_file.write(file_contents)
+
+    data = load_data(source=str(json_file))
+
+    assert data == expected
+
+
+def test_can_load_yaml_url_from_env_var(monkeypatch):
+    with requests_mock.Mocker() as m:
+        m.get("https://test.com/data.yaml", text='{"water": "melon"}')
+        monkeypatch.setenv("CDN_DEFINITIONS_PATH", "https://test.com/data.yaml")
+        data = load_data()
+
+    assert data == {"water": "melon"}
+
+
+def test_can_load_json_url_from_env_var(monkeypatch):
+    with requests_mock.Mocker() as m:
+        m.get("https://test.com/data.json", json='{"green": "bean"}')
```

Comment:
Regarding the earlier comment about wrong use of `json.loads(data.json(), ...)`

I see the issue here in the test.  The `json` here is meant to be *an object which will be serialized as JSON*, not a JSON string.  So this should have been:

```suggestion
        m.get("https://test.com/data.json", json={"green": "bean"})
```

requests_mock accepts the string since strings can also be serialized to JSON, so you've set up a "JSON within JSON" response here, which makes it so that the double-decoding in the code under test happens to work.

Reply:
Whoops! That explains the error that provoked me to use json.loads(data.json()) in the first place. Thanks!